test/plugin/test_out_google_cloud.rb

```diff
@@ -1431,4 +1538,39 @@ def verify_log_entries(n, params, payload_type = 'textPayload')
     end
     assert i == n, "Number of entries #{i} does not match expected number #{n}"
   end
+
+  def underscore(camel_cased_word)
+    camel_cased_word.to_s.gsub(/::/, '/')
+      .gsub(/([A-Z]+)([A-Z][a-z])/, '\1_\2')
+      .gsub(/([a-z\d])([A-Z])/, '\1_\2')
+      .downcase
+  end
+
+  # The caller can optionally provide a block which is called for each entry.
+  def verify_grpc_log_entries(n, params, payload_type = 'textPayload')
+    i = 0
+    @requests_sent.each do |batch|
+      batch.entries.each do |entry|
+        unless payload_type.empty?
+          key = underscore(payload_type)
+          assert entry.send(key), 'LogEntry did not contain expected ' \
+            "#{key} key: " + entry.to_json
```

Comment:
Good point. verify_grpc_log_entries has been changed to first convert the proto to json, then directly call verify_log_entries method.